### ISSUE TITLE
transaction with an unreleased savepoint not properly persisted

### ISSUE THREAD

**Issue by yzchang** [MonetDB/MonetDB#7022]
**Describe the bug**
When committing a transaction with an unreleased `savepoint`, not all information about this transaction is properly persisted. The following problems happen:
1. after the transaction has successfully committed, the test table created in this transaction shows up in the catalogue but can't be found by a `select`.
2. a 2nd transaction which creates a table with the same name succeeds.
3. after the 2nd transaction, both test tables appear in the catalogue. A `select` returns the data of the 2nd test table and a `drop table` only drops the second test table.
These problems do not appear if the savepoint was released.

**To Reproduce**
--- this works, due to the `release savepoint` ---
```
start transaction;
create table savepointtest (id int, primary key(id));
savepoint name1;
insert into savepointtest values(24);
release savepoint name1;
commit;
select * from tables where name = 'savepointtest';
select * from savepointtest;

create table savepointtest (id int, primary key(id));
insert into savepointtest values(42);
select * from tables where name = 'savepointtest';
select * from savepointtest;
drop table savepointtest;
select * from tables where name = 'savepointtest';
```
--- this shows errors ---
```
start transaction;
create table savepointtest (id int, primary key(id));
savepoint name1;
insert into savepointtest values(24);
commit;
select * from tables where name = 'savepointtest';
select * from savepointtest;

create table savepointtest (id int, primary key(id));
insert into savepointtest values(42);
select * from tables where name = 'savepointtest';
-- the 2nd 'savepointtest` is used by SELECT and DROP
select * from savepointtest;
drop table savepointtest;
select * from tables where name = 'savepointtest';
```
**Expected behavior**
The first transaction should be handled correctly. Subsequent creation of a table with the same name should be rejected.

**Software versions**
 - MonetDB version number [v11.39.8 (hg id: 0a97056b6c89)]
 - OS and version: [e.g. Ubuntu 20.04.1]
 - Mercurial source

**Comment by yzchang** [comment]
Test added: sql/test/BugTracker-2020/Tests/transaction_with_unreleased_savepoint.Bug-7022.sql

**Comment by yzchang** [comment]
Fixed by changeset [65bc955a71cd](https://dev.monetdb.org/hg/MonetDB?cmd=changeset;node=65bc955a71cd)
